### DISCUSSION TITLE
MSP Limit on UART Port configuration.

### DISCUSSION THREAD

**Discussion by davemr2** [iNavFlight/inav#7639] [Q&A] [answered]
INAV 2.5.x  Matek.F722.

Every time I try to add configuration for a Matek Optic flow sensor on UART 6. (MSP) It resets other UART Port configurations.
I have UART 1 AS MSP.
GPS on UART 2.
BLE on UART 3. (MSP)

Is there a limit to the number of MSP devices?

**Comment by stronnag** [comment, marked as answer]
https://github.com/iNavFlight/inav/blob/master/docs/Serial.md#constraints
